mycompany-seeq-link-connector-myconnector/src/main/java/com/mycompany/seeq/link/connector/DatasourceSimulator.java

```diff
@@ -1,25 +1,89 @@
 package com.mycompany.seeq.link.connector;
 
 import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
 import java.util.Iterator;
+import java.util.Random;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import com.seeq.link.sdk.utilities.TimeInstant;
+
 
 public class DatasourceSimulator {
-    private boolean connected;
-    private final int tagCount;
-    Duration signalPeriod;
+    public class Element {
```

Comment:
In Java inner classes should almost always be `static`, unless they need to reference an instance of the outer class. I would make all inner classes here `static`.